From 3601a62bac73daaa6d9a7abd095b973c5e473f23 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Sat, 12 Apr 2014 21:20:12 +0300
Subject: [PATCH 172/219] Update min requirements and autotools

Update autotools, fix whitespace issue, change how the version
string is presented with make distcheck, fix --disable-dpms so
it will actually compile, and update the minimum versions for
the 4.12 roadmap.
---
 Makefile.am                          |  4 +--
 configure.ac.in                      | 54 +++++++++++++++++-------------------
 panel-plugins/brightness/Makefile.am |  2 +-
 3 files changed, 28 insertions(+), 32 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 96a6738..bb04112 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,7 +25,7 @@ DISTCLEANFILES = \
 	intltool-merge \
 	intltool-update
 
-distuninstallcheck_listfiles = 						\ 
+distuninstallcheck_listfiles = 						\
 	find . -type f -print | grep -v ./share/icons/hicolor/icon-theme.cache
 
 .PHONY: ChangeLog
@@ -54,4 +54,4 @@ snapshot: dist
 
 snapshot-bz2: dist-bz2
 	mv $(PACKAGE)-$(VERSION).tar.bz2 \
-	$(PACKAGE)-$(VERSION)-r@REVISION@.tar.bz2         
+	$(PACKAGE)-$(VERSION)-r@REVISION@.tar.bz2
diff --git a/configure.ac b/configure.ac
index aef3d68..7a4af87 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,12 +2,12 @@ m4_define([intltool_minimum_version], [0.31])
 m4_define([xfpm_version_major], [1])
 m4_define([xfpm_version_minor], [2])
 m4_define([xfpm_version_micro], [0])
-m4_define([xfpm_version_build], [])
+m4_define([xfpm_version_build], [@REVISION@])
 m4_define([xfpm_version_tag],   [])
-m4_define([xfpm_version], [xfpm_version_major().xfpm_version_minor().xfpm_version_micro()ifelse(xfpm_version_tag(), [git], [xfpm_version_tag().xfpm_version_build()], [xfpm_version_tag()])])
+m4_define([xfpm_version], [xfpm_version_major().xfpm_version_minor().xfpm_version_micro()ifelse(xfpm_version_tag(), [git], [xfpm_version_tag()-xfpm_version_build()], [xfpm_version_tag()])])
 
-AC_INIT([xfce4-power-manager], [xfpm_version], [http://bugzilla.xfce.org/])
-AC_PREREQ(2.50)
+AC_INIT([xfce4-power-manager],[xfpm_version],[http://bugzilla.xfce.org/])
+AC_PREREQ([2.69])
 
 AM_INIT_AUTOMAKE([1.8 dist-bzip2 tar-ustar no-dist-gzip])
 AC_CONFIG_HEADERS([config.h])
@@ -21,9 +21,9 @@ AC_USE_SYSTEM_EXTENSIONS()
 # ===================================================== #
 AC_PROG_CC()
 AM_PROG_CC_C_O()
-AC_PROG_LD()
+LT_PATH_LD([])
 AC_PROG_INSTALL()
-AC_PROG_INTLTOOL([intltool_minimum_version], [no-xml])
+IT_PROG_INTLTOOL([intltool_minimum_version],[no-xml])
 
 # ===================================================== #
 #               Initialize libtool                      #
@@ -53,15 +53,15 @@ XDT_I18N([@LINGUAS@])
 #==================================================== #
 #          Check for required packages 		      #
 #=====================================================#
-m4_define([gtk_minimum_version], [2.12.0])
-m4_define([glib_minimum_version], [2.16.0]) 
-m4_define([dbus_minimum_version], [0.60])
-m4_define([dbus_glib_minimum_version], [0.70])
+m4_define([gtk_minimum_version], [2.24.0])
+m4_define([glib_minimum_version], [2.30.0])
+m4_define([dbus_minimum_version], [1.1])
+m4_define([dbus_glib_minimum_version], [0.84])
 
-m4_define([xfconf_minimum_version], [4.6.0])
-m4_define([libxfce4ui_minimum_version],[4.7.0])
-m4_define([libxfce4util_minimum_version],[4.6.0])
-m4_define([libxfce4panel_minimum_version],[4.6.0])
+m4_define([xfconf_minimum_version], [4.10.0])
+m4_define([libxfce4ui_minimum_version],[4.10.0])
+m4_define([libxfce4util_minimum_version],[4.10.0])
+m4_define([libxfce4panel_minimum_version],[4.10.0])
 
 m4_define([libnotify_minimum_version], [0.4.1])
 m4_define([xrandr_minimum_version], [1.2.0])
@@ -85,8 +85,7 @@ XDT_CHECK_PACKAGE([X11], [x11], [x11_minimum_version])
 #              Polkit?			                #
 #=======================================================#
 AC_ARG_ENABLE([polkit],
-        [AC_HELP_STRING([--disable-polkit],
-                [Do not enable PolicyKit support (default=enabled)])],
+        [AS_HELP_STRING([--disable-polkit],[Do not enable PolicyKit support (default=enabled)])],
         [ac_cv_enable_polkit=$enableval],
         [ac_cv_enable_polkit=polkit])
     AC_MSG_CHECKING([whether to build with polkit support])
@@ -105,17 +104,14 @@ AM_CONDITIONAL([ENABLE_POLKIT], [test "x$polkit" = "xyes"])
 #=======================================================#
 DPMS_LIBS=
 AC_ARG_ENABLE([dpms],
-AC_HELP_STRING([--enable-dpms], [use dpms extension])
-AC_HELP_STRING([--disable-dpms], [don't use dpms extension]),
+AS_HELP_STRING([--enable-dpms],[use dpms extension])
+AS_HELP_STRING([--disable-dpms],[don't use dpms extension]),
   [], [enable_dpms=yes])
-have_dpms="no"
 if test x"$enable_dpms" = x"yes"; then
-     AC_CHECK_LIB([Xext], [DPMSQueryExtension],
-        [ have_dpms="yes"
-          DPMS_LIBS=" -lXext -lX11"
-          AC_DEFINE([HAVE_DPMS], [1], [Define to enable dpms])
-        ],[])
+  AC_DEFINE([HAVE_DPMS], [1], [Define to enable dpms])
 fi
+
+AC_CHECK_LIB([Xext], [DPMSQueryExtension], [DPMS_LIBS=" -lXext -lX11"],[])
 AC_SUBST([DPMS_LIBS])
 
 #=======================================================#
@@ -134,7 +130,7 @@ fi
 #=======================================================#
 #              Network Manager support                  #
 #=======================================================#
-AC_ARG_ENABLE([network-manager], AC_HELP_STRING([--disable-network-manager], [Disable network manager support]),
+AC_ARG_ENABLE([network-manager], AS_HELP_STRING([--disable-network-manager],[Disable network manager support]),
   [ac_network_manager=$enableval], [ac_network_manager=yes])
 AC_MSG_CHECKING([whether to build with network manager support.])
 if test "x$ac_network_manager = xyes"; then
@@ -146,8 +142,7 @@ AC_MSG_RESULT([$ac_network_manager])
 #                 Panel plugins ?                       #
 #=======================================================#
 AC_ARG_ENABLE([panel_plugins],
-	[AC_HELP_STRING([--disable-panel-plugins],
-		[Do not build panel plugins (default=enabled)])],
+	[AS_HELP_STRING([--disable-panel-plugins],[Do not build panel plugins (default=enabled)])],
 	[ac_cv_enable_panel_plugins=$enableval],
 	[ac_cv_enable_panel_plugins=yes])
 if test "x$ac_cv_enable_panel_plugins" = "xno"; then
@@ -177,7 +172,7 @@ AC_MSG_CHECKING([PLATFORM_LDFLAGS])
 AC_MSG_RESULT([$PLATFORM_LDFLAGS])
 AC_SUBST([PLATFORM_LDFLAGS])
 
-AC_OUTPUT([
+AC_CONFIG_FILES([
 Makefile
 libdbus/Makefile
 common/Makefile
@@ -207,11 +202,12 @@ data/icons/scalable/actions/Makefile
 data/interfaces/Makefile
 po/Makefile.in
 ])
+AC_OUTPUT
 
 echo "prefix:			${prefix}"
 echo "xdg autostart:		${sysconfdir}/xdg/autostart"
 echo "POLKIT:			${polkit}"
-echo "DPMS:			${have_dpms}"
+echo "DPMS:			${enable_dpms}"
 echo "Network manager:	${ac_network_manager}"
 echo "Build panel plugins:	${build_panel_plugins}"
 echo "Debug:			${enable_debug}"
diff --git a/panel-plugins/brightness/Makefile.am b/panel-plugins/brightness/Makefile.am
index a89ca8b..65e0f5c 100644
--- a/panel-plugins/brightness/Makefile.am
+++ b/panel-plugins/brightness/Makefile.am
@@ -1,4 +1,4 @@
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-DG_LOG_DOMAIN=\"xfce4-brightness-plugin\" \
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\" \
-- 
1.9.3

